fix: ensure handleChangeCountry is stable

JanStevens opened the pull request:
Fixes https://github.com/viclafouch/mui-tel-input/issues/31

A bit more changes than I have hoped but this allows to use keyboard accessibility to jump to the countries starting with the letter `b` when pressing `b` on your keyboard.

I also refactored the `useState` "hack" to `useMemo` else I had issues with changing countries clearing the existing input.

viclafouch commented on src/index.tsx:
```diff
@@ -86,19 +86,16 @@ const MuiTelInput = React.forwardRef(
-        focusInputElement()
-      }
-    }
+    const handleChangeCountry = React.useCallback(
```
I don't understand why this is a `useCallback` as the definition of `onCountryChange` is always different.

What the purpose of memoizing the function ?

JanStevens replied:
To be honested, I also thought there is no point in the `useState` and `useCallback`. The `useState` hack causes other issues and the `useCallback` seems to be unneeded as it always will rerender

viclafouch commented on src/components/FlagsMenu/FlagsMenu.tsx:
```diff
@@ -29,6 +34,32 @@ const FlagsMenu = (props: FlagsMenuProps) => {
+  }, [
+    continents,
+    displayNames,
+    excludedCountries,
```
not sure about removing the old trick, because here, your value as deps are always differents. `excludedCountries` is always recreated on every render, same for the rest of the deps :/

JanStevens replied:
See previous comment, I think TBH that all this "optimisation" can be removed. I don't think there is any issue in rerendering

JanStevens commented:
@viclafouch I've updated the PR, removed the unneeded useCallbacks. A small optimisation I did is passing the `countryName` directly too `FlagMenuItem` instead of the `displayNames` object. I think that will do? Let me know if you have any other remarks

ja153903 commented:
What's the likelihood that this PR goes through? Keyboard accessibility is quite important to our product and would love to see this resolved.
